### PR TITLE
Fix toHaveStyle not to return true when shouldn't

### PR DESCRIPTION
**Pull request by CarlaTeo**
**What**:
[As the docs says](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/style), the `style` property of an HTMLElement should not be directly assigned:
```
Styles should not be set by assigning a string directly to the style property (as in elt.style = "color: blue;"), since it is considered read-only, as the style attribute returns a CSSStyleDeclaration object which is also read-only. Instead, styles can be set by assigning values to the properties of style.
```
This probably is related to the issue: https://github.com/gnapse/jest-dom/issues/68

**Why**:
By directly assigning style, whenever a nonexistent style was passed to `getStyleDeclaration`, it would return an empty object, making `toHaveStyle` always return true in these cases.

**How**:
Since the goal was to obtain a key value pair of style properties, `getStyleDeclaration` was changed to parse the string and obtain an object.
That object could be returned, but the HTMLElement style attribution was kept to normalize colors.

**Checklist**:

* [ ] Documentation  "N/A"
* [ x] Tests
* [ ] Updated Type Definitions  "N/A"
* [ x] Ready to be merged 
* [ x] Added myself to contributors table

### CODE REVIEW

File: src/to-have-style.js [hunk 1]
```diff
@@ -4,14 +4,28 @@ import chalk from 'chalk'
...
+    .split(';')
+    .map(item => item.trim())
+    .filter(e => e)
+
+  const desiredStyles = keyValues.reduce((acc, cur) => {
+    const key = cur.substr(0, cur.indexOf(':')).trim()
+    const val = cur.substr(cur.indexOf(':') + 1).trim()
+    acc[key] = val
+    return acc
+  }, {})
```
[the remark this answers is not in the document]
Comment (gnapse) [reply src/to-have-style.js]: Actually, I recalled that we used to parse the entire css with this library initially. I tracked down when it stopped being this way, and I found that this was changed in #57. Maybe @smacpherson64 could help us recall why this was changed.

Also @CarlaTeo, checkout [this diff from that PR](https://github.com/gnapse/jest-dom/pull/57/files#diff-4e1129d165c9efdd245d6f1726948bbb) to see how this used to be before #57.

**Comment by kevinSuttle** [comment]
I'm seeing this fail with both single and compound styles.

```js
const {getByText} = render(<Button label="Default" />);
expect(getByText('Default')).toHaveStyle(
      'backgroundcolor: LITERALLY ANYTHING'
    );
// CLI: 👍 ✅  
```

**Comment by gnapse** [comment]
> I'm seeing this fail with both single and compound styles.
> 
> ```js
> const {getByText} = render(<Button label="Default" />);
> expect(getByText('Default')).toHaveStyle(
>       'backgroundcolor: LITERALLY ANYTHING'
>     );
> // CLI: 👍 ✅  
> ```

@kevinSuttle I'm not sure I understand your comment.

What are you seeing fail? The version of this library in this PR? Or are you referring to this library's currently stable release?

What about that thumbs-up and checkmark at the end of your comment alongside the "CLI" word? What does that mean? Because it seems to convey that things are working, not failing.

**Comment by dmwyatt** [comment]
@CarlaTeo Any chance you can look into getting this fixed up?

**Comment by nealoke** [comment]
Please merge this in, now any test with a invalid value just passes... 😢

**Comment by weyert** [comment]
You can use Cypress for this? See: https://docs.cypress.io/guides/references/assertions.html#CSS
My understanding was that jsdom doesn't interpret the styling? That's why I am using Cypress to assert these kind of things.

**Comment by CarlaTeo** [comment]
Sorry for the delay, people. 🙈

File: src/__tests__/to-have-style.js [hunk 2]
```diff
@@ -96,4 +96,16 @@ describe('.toHaveStyle', () => {
...
+
+  test('handles nonexistent styles', () => {
+    const {container} = render(`
+          <div class="label" style="background-color: blue; height: 100%">
+            Hello World
+          </div>
+        `)
+
+    expect(container.querySelector('.label')).not.toHaveStyle(
+      'whatever: anything',
```
[the remark this answers is not in the document]
Comment (CarlaTeo) [reply src/__tests__/to-have-style.js]: I've added tests regarding both the compound statement and the rgba comment. 
I'm not sure I got the issues right, but it seems even without my changes the tests were passing...🤔

File: src/to-have-style.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (CarlaTeo) [reply src/to-have-style.js]: Done. Thank you for the guidance.

File: src/__tests__/to-have-style.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (CarlaTeo) [reply src/__tests__/to-have-style.js]: Well, it was...

I was working with HEAD in 3da3666cb25277d7fbef7826a6e928ccc079a455, but when I rebased with master before pushing, it seems it no longers work... -.-

**Comment by CarlaTeo** [comment]
It indeed does not solves #68 but still brings some improvement.

For future TDD on the issue 68 I left a commented test.

**Comment by klaaspieter** [comment]
I'm having issues with `toHaveStyle` as well. Anything I can do to get this merged?

**Comment by gnapse** [comment]
@klaaspieter my apologies for not having followed up on this as I said I would. Does this PR as it is covers the issues you're having with `.toHaveStyles`? I'm going to be merging it now, I just wanted to make sure the issues you mention are the ones this PR solves.

**Comment by klaaspieter** [comment]
@gnapse I didn't test it with my own changes but I think at a minimum this PR fixes a false positive. I was trying to test for this:

```javascript
expect(element).toHaveStyle('-webkit-overflow-scrolling: touch')
```

If I recall correctly before this PR this would falsely claim be considered true  because parsing the expected CSS resulted in `{}` and looking for the `-webkit-overflow-scrolling` key on `element` also resulted in `{}` (I think because JSDom doesn't support the vendor prefix it's never added to the element). 

Conversely `expect(element).not.toHaveStyle('-webkit-overflow-scrolling: touch')` failed with an error saying that there is no difference between the expected and actual result because they're both `{}`. In this case the `.not.toHaveStyle` should actually pass.

I _think_ this PR fixes that problem but I'd have to test it again to make sure.

**Comment by gnapse** [comment]
I think this PR may have introduced a problem. This now does not pass:

```jsx
const { container } = render(<div style={{ border: 'none' }} />)
expect(container.firstChild).toHaveStyle('border: none')
```

Could it be? If so, what could we do about it?

**Comment by CarlaTeo** [comment]
> I think this PR may have introduced a problem. This now does not pass:
> 
> ```js-jsx
> const { container } = render(<div style={{ border: 'none' }} />)
> expect(container.firstChild).toHaveStyle('border: none')
> ```
> 
> Could it be? If so, what could we do about it?

I am not sure about what problem you meant...I tried the following test:
```
const { container } = render(`<div class="label" style="border: 'none'" />`);
expect(container.querySelector('.label')).toHaveStyle('border: none')
```
And it does pass.


In fact, I realized this PR solves the case of nonexistant keys, but not values, since:
```
const { container } = render(<div style={{ border: 'none' }} />)
expect(container.firstChild).toHaveStyle('border: anything')
```
Still passes...

**Comment by gnapse** [comment]
Yeah, never mind, I had updated an old project to the latest version of jest-dom, and now a test similar to what I depicted above does not pass, failing at this style rule precisely. But I cannot reproduce in a minimal react app repo. False alarm.

**Pull request merged**